FrontEnd/src/components/ProfileDetail/MainInfo/ProfileDetailNavBar.jsx

```diff
@@ -10,38 +14,66 @@ const MENU_LINKS = {
     'cooperation': 'Формат співпраці',
 };
 
-function ProfileDetailNavBar() {
-
+function ProfileDetailNavBar({ data }) {
   const { hash } = useLocation ();
+  const navigate = useNavigate ();
+  const { dataInComponents } = useContext(DataContext);
+  const [activeLink, setActiveLink] = useState('');
+
+  useEffect(() => {
+    if (hash) {
+      setActiveLink(hash.substring(1));
+    } else {
+      setActiveLink(
+        data.is_registered && dataInComponents.includes('about-company') ? 'about-company' :
+        data.is_startup && dataInComponents.includes('startup') ? 'startup' : ''
+      );
+    }
+  }, [hash, data.is_registered, data.is_startup, dataInComponents]);
+
+  useEffect(() => {
+    navigate(hash.pathname, {replace: true});
+  }, [navigate, hash.pathname]);
 
-  return  (
-    <div className={classes['navbar-menu']}>
-      {Object.entries(MENU_LINKS).map(([link, label]) => (
-        <div key={link} className={classes['navbar-menu__block']}>
-          <div className={classes['navbar-menu__item']}>
-            <HashLink
-              smooth
-              to={`#${link}`}
-              className={
-                `#${link}` === hash
-                  ? `${classes['active-link']}`
-                  : `${classes['inactive-link']}`
-              }
-            >
-              {label}
-            </HashLink>
-          </div>
-          <div
-            className={
-              `#${link}` === hash
-                ? `${classes['active-devider']}`
-                : `${classes['inactive-devider']}`
-            }
-          />
-        </div>
-      ))}
+  return (
+    <div>
+      <div className={classes['navbar-menu']}>
+        {Object.entries(MENU_LINKS).map(
+          ([link, label]) =>
+            dataInComponents.includes(link) && (
```

Comment:
Since we store links could we provide a better name for the context that describes it? Like `activeTabs` or `activeLinks`. Same for the context provider

Reply:
Done